### PR TITLE
Add Rules for Import Ordering and Sorting

### PR DESCRIPTION
**Pull request by disufi**
References: https://autoricardo.atlassian.net/browse/CAR-4179

### Description

By example, let's say we have the following import statements:

```ts
import { fetchFoo } from "@carforyou/api-client"
import { NextPage } from "next"
import React from "react"
import dynamic from "next/dynamic"
import { useTranslation } from "~/i18n"
import { bFoo, aFoo } from "~/lib/foo"
import { bBar, aBar } from "~/lib/bar"
import image from "./image.png"
import locales from "~/i18n/locales"
```

Once you run the auto-fix, you get:

```ts
import React from "react"
import dynamic from "next/dynamic"
import { NextPage } from "next"
import { fetchFoo } from "@carforyou/api-client"

import { aFoo, bFoo } from "~/lib/foo"
import { aBar, bBar } from "~/lib/bar"
import locales from "~/i18n/locales"
import { useTranslation } from "~/i18n"

import image from "./image.png"
```

Imports get grouped following this pattern:

```ts
["builtin", "external"],
"internal",
["parent", "sibling"],
["index", "object"]
``` 

Between groups a newline is required and newlines are allowed inside groups.

There is a custom path group that matches `~/**` as `internal` such as `~lib/**` or `~components/*` to wrap these together.

### Notes

We could go some steps further and specify further custom groups to e.g. have separate grouping for `@carforyou/**` and each a group for e.g. `~components/**` or `~lib/**` but I think that might be too specific and restrictive, awaiting your thoughts on this.

### CODE REVIEW

**Comment by Averethel** [comment]
It's a solid starting point. What we do when ordering by hand so far is:
- external packages (preferably grouped together e.g. `next` close to `next/link`)
- internal packages (`@carforyou/*`)
- imports from `~/lib` 
- hooks and contexts (`components/hooks`, `components/contexts`)
- asset components (`assets/icons`)
- other components 
- css modules

If we could continue in that direction that would be great.

**Review by ralucacril — APPROVED** [review]

**Comment by disufi** [comment]
@Averethel tried to incorporate your feedback, some notes:

> external packages (preferably grouped together e.g. next close to next/link)

Done, they are sorted alphabetically, next will be grouped by default.

> internal packages (@carforyou/*)

Done.

> imports from ~/lib

Done.

> hooks and contexts (components/hooks, components/contexts)

Done, first all hooks, then all contexts.

> asset components (assets/icons)

Done, I used the match all `~assets/**` here.

> other components

Done.

> css modules

These are relative and will be grouped in this order as well by `["parent", "sibling"]`.

---

Could you please review as well? Thanks!

**Review by lkappeler — COMMENTED** [review]
I don't have any preference on this so I am not the right one to review the change but thanks a lot for tackling it, so we don't need to discuss the order of imports in code reviews 😍

File: index.js [hunk 1]
```diff
@@ -105,5 +105,73 @@ module.exports = {
...
+            group: "internal",
+            position: "after",
+          },
+          {
+            pattern: "~/components/contexts/**",
+            group: "internal",
+            position: "after",
+          },
+          {
+            pattern: "~/assets/**",
```
Comment (masone) [inline index.js:140]: some of the patterns here are project-specific, while this package has been pretty generic so far. I would not want to couple project-specific knowledge to this package

one option would be to not enforce sorting for those. I personally do not care much tbh. if external imports are at the top, I'm fine

another option is to move the project-specific rules to the projects themselves. or at least move them to the right file (we expose multiple configs. one per project-type)

I'm open for a discussion here, but would rather accept some tradeoffs than tight coupling

Comment (disufi) [reply index.js:140]: I totally concur that we should ideally separate / extract some of the rules and put them nearer to the respective projects. I think the only thing we have to do is move the `pathGroups` to the project specific es lint config, either in this repo or actually to the specific repos, I'll check the options and adapt this PR accordingly.

Comment (disufi) [reply index.js]: @masone Do you think moving some of the more specific rules to the next config is sufficient? Or do the next projects differ too much and we'd have to move it to the project config (respective repo) itself?

**Comment by disufi** [comment]
@Averethel After a discussion with @masone and @lkappeler we decided to drop the project specific path groups (ordering) and go with a less opinionated solution which we would extend in a follow up easily on the respective projects itself.

**Pull request merged**